contracts/manifold/burnredeem/ERC721BurnRedeem.sol

```diff
@@ -27,63 +30,90 @@ contract ERC721BurnRedeem is BurnRedeemCore, IERC721BurnRedeem {
+            creatorContractVersion = uint8(version);
+        } catch {}
+        _initialize(creatorContractAddress, creatorContractVersion, instanceId, burnRedeemParameters);
+        _identicalTokenURI[creatorContractAddress][instanceId] = identicalTokenURI;
     }
 
     /**
      * @dev See {IERC721BurnRedeem-updateBurnRedeem}.
      */
     function updateBurnRedeem(
         address creatorContractAddress,
-        uint256 index,
+        uint256 instanceId,
         BurnRedeemParameters calldata burnRedeemParameters,
         bool identicalTokenURI
     ) external creatorAdminRequired(creatorContractAddress) {
-        _update(creatorContractAddress, index, burnRedeemParameters);
-        _identicalTokenURI[creatorContractAddress][index] = identicalTokenURI;
+        _update(creatorContractAddress, instanceId, burnRedeemParameters);
+        _identicalTokenURI[creatorContractAddress][instanceId] = identicalTokenURI;
     }
 
     /**
      * See {IERC721BurnRedeem-updateTokenURI}.
      */
     function updateTokenURI(
         address creatorContractAddress,
-        uint256 index,
+        uint256 instanceId,
         StorageProtocol storageProtocol,
         string calldata location,
         bool identicalTokenURI
     ) external override creatorAdminRequired(creatorContractAddress) {
-        BurnRedeem storage burnRedeemInstance = _getBurnRedeem(creatorContractAddress, index);
+        BurnRedeem storage burnRedeemInstance = _getBurnRedeem(creatorContractAddress, instanceId);
         burnRedeemInstance.storageProtocol = storageProtocol;
         burnRedeemInstance.location = location;
-        _identicalTokenURI[creatorContractAddress][index] = identicalTokenURI;
-        emit BurnRedeemUpdated(creatorContractAddress, index);
+        _identicalTokenURI[creatorContractAddress][instanceId] = identicalTokenURI;
+        emit BurnRedeemLib.BurnRedeemUpdated(creatorContractAddress, instanceId);
     }
 
     /** 
      * Helper to mint multiple redeem tokens
      */
-    function _redeem(address creatorContractAddress, uint256 index, BurnRedeem storage burnRedeemInstance, address to, uint32 count) internal override {
+    function _redeem(address creatorContractAddress, uint256 instanceId, BurnRedeem storage burnRedeemInstance, address to, uint32 count) internal override {
         if (burnRedeemInstance.redeemAmount == 1 && count == 1) {
             ++burnRedeemInstance.redeemedCount;
-            uint256 newTokenId = IERC721CreatorCore(creatorContractAddress).mintExtension(to);
-            _redeemTokens[creatorContractAddress][newTokenId] = RedeemToken(uint224(index), burnRedeemInstance.redeemedCount);
-            emit BurnRedeemMint(creatorContractAddress, index, newTokenId, 1);
+            uint256 newTokenId;
+            if (burnRedeemInstance.contractVersion >= 3) {
+                uint80 tokenData = uint56(instanceId) << 24 | burnRedeemInstance.redeemedCount;
```

Comment:
where can i learn the purpose of `tokenData`? idempotency token for mints?

`redeemedCount` is `uint32`

Reply:
no, it doesn't matter it's a uint32.  Will still OR properly and take the lower 32 bits.

tokenData is just immutable data stored with the token when minted that's 'free' due to struct packing (the struct is the mutable ownerAddress and immutable tokenData)